packages/analytics-core/src/analytics.js

```diff
@@ -62,27 +54,63 @@ export default class AvAnalytics {
   }
 
   handleEvent = event => {
-    if (
-      isModifiedEvent(event) ||
-      (event.type === 'click' && !isLeftClickEvent(event)) ||
-      !isValidEventTypeOnTarget(event)
-    ) {
+    if (this.invalidEvent(event)) {
       return;
     }
+
     const target = event.target || event.srcElement;
-    const analyticAttrs = getAnalyticAttrs(target);
+    const { path } = event;
+    let analyticAttrs = {};
+
+    if (this.recursive) {
+      // Reverse the array so we pull attributes from top down
+      path.reverse().forEach(pth => {
+        const attrs = this.getAnalyticAttrs(pth);
+
+        analyticAttrs = { ...analyticAttrs, ...attrs };
+
+        analyticAttrs.action = analyticAttrs.action || event.type;
+      });
+    } else {
+      analyticAttrs = this.getAnalyticAttrs(target);
+      analyticAttrs.elemId =
+        analyticAttrs.elemId ||
+        target.getAttribute('id') ||
+        target.getAttribute('name');
+      analyticAttrs.action = analyticAttrs.action || event.type;
+    }
 
     if (!Object.keys(analyticAttrs).length > 0) {
       return;
     }
-    analyticAttrs.elemId =
-      analyticAttrs.elemId ||
-      target.getAttribute('id') ||
-      target.getAttribute('name');
-    analyticAttrs.action = analyticAttrs.action || event.type;
     this.trackEvent(analyticAttrs);
   };
 
+  validEvent = event =>
+    isModifiedEvent(event) ||
+    (event.type === 'click' && !isLeftClickEvent(event)) ||
+    !isValidEventTypeOnTarget(event);
+
+  getAnalyticAttrs = elem => {
+    if(!elem.attributes) {
+      return {};
+    }
+
+    const attrs = [...elem.attributes];
+    const analyticAttrs = {};
+
+    if (elem.nodeType === 1) {
+      for (let i = attrs.length - 1; i >= 0; i--) {
+        const { name } = attrs[i];
+        if (name.indexOf(`${this.attributePrefix}-`) === 0) {
+          const camelName = camelCase(name.slice(15));
```

Comment:
Looks like that slice is based on the default prefix, should change with the option.